### PR TITLE
save style for mesh layer and refactoring for raster and vector widge…

### PR DESCRIPTION
**Pull request by vcloarec**
…t properties

## Description

[Replace this with some text explaining the rationale and details about this pull request]

### CODE REVIEW

File: src/app/mesh/qgsmeshlayerproperties.cpp [hunk 1]
```diff
@@ -91,6 +93,18 @@ QgsMeshLayerProperties::QgsMeshLayerProperties( QgsMapLayer *lyr, QgsMapCanvas *
   if ( !mMeshLayer->styleManager()->isDefault( mMeshLayer->styleManager()->currentStyle() ) )
     title += QStringLiteral( " (%1)" ).arg( mMeshLayer->styleManager()->currentStyle() );
   restoreOptionsBaseUi( title );
+
+  QPushButton *btnStyle = new QPushButton( tr( "Style" ) );
```
Comment (PeterPetrik) [inline src/app/mesh/qgsmeshlayerproperties.cpp:97]: cannot be the push button part of the UI file?

File: src/app/mesh/qgsmeshlayerproperties.cpp [hunk 2]
```diff
@@ -189,6 +203,108 @@ void QgsMeshLayerProperties::addDataset()
   }
 }
 
+void QgsMeshLayerProperties::loadDefaultStyle()
+{
+  bool defaultLoadedFlag = false;
+  QString myMessage = mMeshLayer->loadDefaultStyle( defaultLoadedFlag );
+  //reset if the default style was loaded OK only
```
Comment (PeterPetrik) [inline src/app/mesh/qgsmeshlayerproperties.cpp]: missing whitespace in the comment

File: src/app/mesh/qgsmeshlayerproperties.cpp [hunk 3]
```diff
@@ -189,6 +203,108 @@ void QgsMeshLayerProperties::addDataset()
...
+  if ( defaultLoadedFlag )
+  {
+    syncToLayer();
+  }
+  else
+  {
+    //otherwise let the user know what went wrong
```
Comment (PeterPetrik) [inline src/app/mesh/qgsmeshlayerproperties.cpp]: missing whitespace in the comment

File: src/app/mesh/qgsmeshlayerproperties.cpp [hunk 4]
```diff
@@ -189,6 +203,108 @@ void QgsMeshLayerProperties::addDataset()
...
+
+  // a flag passed by reference
+  bool defaultSavedFlag = false;
+  // after calling this the above flag will be set true for success
+  // or false if the save operation failed
+  QString myMessage = mMeshLayer->saveDefaultStyle( defaultSavedFlag );
+  if ( !defaultSavedFlag )
+  {
+    //let the user know what went wrong
+    QMessageBox::information( this,
```
Comment (PeterPetrik) [inline src/app/mesh/qgsmeshlayerproperties.cpp:237]: should it be at least warning if the saving of the style failed?

File: src/app/mesh/qgsmeshlayerproperties.h [hunk 5]
```diff
@@ -59,7 +60,16 @@ class APP_EXPORT QgsMeshLayerProperties : public QgsOptionsDialogBase, private U
     void changeCrs( const QgsCoordinateReferenceSystem &crs );
     //! Associate dataset to the mesh layer
     void addDataset();
+    //! Load the default style when appropriate button is pressed.
```
Comment (PeterPetrik) [inline src/app/mesh/qgsmeshlayerproperties.h]: Loads, remove extra . at the end

File: src/app/mesh/qgsmeshlayerproperties.h [hunk 6]
```diff
@@ -59,7 +60,16 @@ class APP_EXPORT QgsMeshLayerProperties : public QgsOptionsDialogBase, private U
...
+    void loadDefaultStyle();
+    //! Save the default style when appropriate button is pressed.
```
Comment (PeterPetrik) [inline src/app/mesh/qgsmeshlayerproperties.h]: Saves, remove extra . at the end

File: src/app/mesh/qgsmeshlayerproperties.h [hunk 7]
```diff
@@ -59,7 +60,16 @@ class APP_EXPORT QgsMeshLayerProperties : public QgsOptionsDialogBase, private U
...
+    void saveDefaultStyle();
+    //! Load a saved style when appropriate button is pressed.
```
Comment (PeterPetrik) [inline src/app/mesh/qgsmeshlayerproperties.h]: Loads , remove extra . at the end

File: src/app/mesh/qgsmeshlayerproperties.h [hunk 8]
```diff
@@ -59,7 +60,16 @@ class APP_EXPORT QgsMeshLayerProperties : public QgsOptionsDialogBase, private U
...
+    void loadStyle();
+    //! Save a style when appriate button is pressed.
```
Comment (PeterPetrik) [inline src/app/mesh/qgsmeshlayerproperties.h]: Saves, remove extra . at the end

File: src/gui/qgsmaplayerstyleguiutils.h [hunk 9]
```diff
@@ -51,6 +51,13 @@ class GUI_EXPORT QgsMapLayerStyleGuiUtils : public QObject
      */
     void addStyleManagerActions( QMenu *m, QgsMapLayer *layer );
 
+    /**
+     * \brief removes extra separators from the menu
+     */
```
Comment (PeterPetrik) [inline src/gui/qgsmaplayerstyleguiutils.h:58]: \since QGIS 3.14 ... or add as private method?

**Review by PeterPetrik — COMMENTED** [review]
looks good. please add some tests too

File: src/app/mesh/qgsmeshlayerproperties.cpp (hunk 1 above)
Comment (vcloarec) [reply src/app/mesh/qgsmeshlayerproperties.cpp:97]: no, as this button take the place of the reset button on the box button, with it in ui files, it leads to have the text "reset" too and picture.
With this implementation, we don"t need to replace the text and remove the image.

**Pull request closed without merging**